plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/validator/FileDecorator.java

```diff
@@ -0,0 +1,250 @@
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.developerstudio.eclipse.gmf.esb.diagram.validator;
+
+import org.eclipse.jface.viewers.IDecoration;
+import org.eclipse.jface.viewers.ILightweightLabelDecorator;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.wso2.developerstudio.eclipse.gmf.esb.diagram.custom.deserializer.Deserializer;
+import org.wso2.developerstudio.eclipse.gmf.esb.diagram.custom.deserializer.Deserializer.DeserializeStatus;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Scanner;
+
+import org.eclipse.core.internal.resources.Folder;
+import org.eclipse.core.internal.resources.Project;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.resource.ImageDescriptor;
+
+public class FileDecorator extends LabelProvider implements ILightweightLabelDecorator {
+
+    private static final String ICON = "/icons/Symbol-warning-small.png";
+    private static final String API = "api";
+    private static final String ENDPOINTS = "endpoints";
+    private static final String INBOUND_ENDPOINT = "inbound-endpoints";
+    private static final String LOCAL_ENTRY = "local-entries";
+    private static final String MS_PROCESSOR = "message-processors";
+    private static final String MS_STORE = "message-stores";
+    private static final String PROXY = "proxy-services";
+    private static final String SEQUENCES = "sequences";
+    private static final String TASKS = "tasks";
+    private static final String TEMPLATES = "templates";
+    private static final String SRC = "src";
+    private static final String MAIN = "main";
+    private static final String SYNAPSE_CONFIG = "synapse-config";
+    private static final String[] ARTIFACTS = { API, ENDPOINTS, INBOUND_ENDPOINT, LOCAL_ENTRY, MS_PROCESSOR, MS_STORE,
+	    PROXY, SEQUENCES, TASKS, TEMPLATES };
+    private static final ImageDescriptor IMAGE_DESCRIPTOR = ImageDescriptor.createFromFile(FileDecorator.class, ICON);
+
+    private static Deserializer deserializer = null;
+
+    /**
+     * Method which is triggered by the org.eclipse.ui.decorators extension.
+     */
+    @Override
+    public void decorate(Object element, IDecoration decoration) {
+
+	if (element instanceof IFile) {
+
+	    IFile iFile = (IFile) element;
+	    String fileName = iFile.getName();
+
+	    if (!fileName.equals("pom.xml") && iFile.getFileExtension().equals("xml")
+		    && getProjectType(iFile.getFullPath().toString(), iFile.getProject().getName(), fileName)) {
+
+		try {
+		    String source = new Scanner(iFile.getContents()).useDelimiter("\\A").next();
+		    if (!isValid(source)) {
+			addDecorator(decoration);
+		    }
+
+		} catch (CoreException e) {
+		    // ignore
```

Comment:
Don't we need to throw/handle the exception?

Reply:
Since this is a registered extension, we cannot throw the exception. But, we can return false at https://github.com/wso2/devstudio-tooling-esb/pull/415/files#diff-b4b03c371203344cec40f0c9e26d883eR202 to indicate the error. Will update, thanks :)